src/pages/AddressInfoPage/AddressTransactionRow.tsx

```diff
@@ -0,0 +1,192 @@
+
+  alphAmount = alphAmount < 0 ? alphAmount * BigInt(-1) : alphAmount
+
+  const infoType = isConsolidationTx(t) ? 'move' : getDirection(t, addressHash)
+  const { amountTextColor, amountSign, Icon, iconColor, iconBgColor } = useTransactionUI(infoType)
+
+  const renderOutputAccounts = () => {
+    if (!t.outputs) return
+    // Check for auto-sent tx
+    if (t.outputs.every((o) => o.address === addressHash)) {
+      return <AddressLink key={addressHash} address={addressHash} maxWidth="250px" />
+    } else {
+      const outputs = _(t.outputs.filter((o) => o.address !== addressHash))
+        .map((v) => v.address)
+        .uniq()
+        .value()
+
+      return (
+        <AccountsSummaryContainer>
+          <AddressLink address={outputs[0]} maxWidth="250px" />
+          {outputs.length > 1 && ` (+ ${outputs.length - 1})`}
+        </AccountsSummaryContainer>
+      )
+    }
+  }
+
+  const renderInputAccounts = () => {
+    if (!t.inputs) return
+    const inputs = _(t.inputs.filter((o) => o.address !== addressHash))
+      .map((v) => v.address)
+      .uniq()
+      .value()
+
+    return inputs.length > 0 ? (
+      <AccountsSummaryContainer>
+        {inputs[0] && <AddressLink address={inputs[0]} maxWidth="250px" />}
+        {inputs.length > 1 && ` (+ ${inputs.length - 1})`}
+      </AccountsSummaryContainer>
+    ) : (
+      <BlockRewardLabel>Block rewards</BlockRewardLabel>
+    )
+  }
+
+  const directionIconSize = 15
+
+  return (
+    <>
+      <TableRow key={t.hash} isActive={detailOpen} onClick={toggleDetail}>
+        <IconContainer style={{ backgroundColor: iconBgColor }}>
+          <Icon size={directionIconSize} strokeWidth={2} color={iconColor} />
+        </IconContainer>
+
+        <TightLink to={`/transactions/${t.hash}`} text={t.hash} maxWidth="120px" />
+
+        {(t.timestamp && <Timestamp timeInMs={t.timestamp} />) || '-'}
+
+        <Badge
+          type="neutral"
+          content={infoType === 'move' ? 'Moved' : infoType === 'out' ? 'To' : 'From'}
+          floatRight
+          minWidth={60}
+        />
+
+        {infoType === 'move' || infoType === 'out' ? renderOutputAccounts() : renderInputAccounts()}
+        <AmountCell color={amountTextColor}>
+          {amountSign}
+          <Amount value={alphAmount} />
+        </AmountCell>
+        <DetailToggle isOpen={detailOpen} />
+      </TableRow>
+      <TableDetailsRow openCondition={detailOpen}>
+        <AnimatedCell colSpan={7}>
+          <Table>
+            <TableHeader headerTitles={['Inputs', '', 'Outputs']} columnWidths={['', '50px', '']} />
+            <TableBody>
+              <TableRow>
+                <div>
+                  {t.inputs && t.inputs.length > 0 ? (
+                    t.inputs.map(
+                      (input, i) =>
+                        input.address && (
+                          <AddressLink
+                            key={i}
```

Comment:
nit: the index is not a good key, maybe there's another unique value in the `input` object?

Reply:
Used txHashRef